### PR TITLE
Feature/130 complete file upload page

### LINKED ISSUE #130: Complete the implementation of the File Upload Page
**Tasks:**

- Develop a File Upload Page component based on the finalized design structure.
- Implement TypeScript for static typing throughout the File Upload Page component.
- Define TypeScript interfaces for file upload-related data structures.
- Implement logic to handle user file uploads.
- Validate file types and sizes to ensure compatibility with the processing system.
- Connect the File Upload Page to the backend API for storing uploaded files.
- Implement error handling for file upload requests.
- Apply styles to the File Upload Page component using CSS Modules or Styled Components.
- Ensure styles align with the approved design and maintain consistency across the application.
- Implement responsive design to ensure the File Upload Page is accessible on various screen sizes.
- Test and adjust styles for mobile and desktop views.
- Implement interactive features, such as displaying upload progress or providing feedback on successful uploads.
- Ensure a smooth and intuitive user experience.
- Write unit tests for the File Upload Page component using Jest and React Testing Library.
- Conduct thorough testing to identify and address any issues.

**Note:**

- As usual please ensure extensive use of comments, Jest tests, error handling and validation where applicable.
- The estimate for this is 4 days

### PR DESCRIPTION
**Pull request by redwing14e**
Resolves #130 

Description

Allows for uploading files

### CODE REVIEW

**Review by ralphgregorio — APPROVED** [review]
paired and looks good

**Review by cameronbeattie306 — APPROVED** [review]
looks good

File: frontend/src/pages/FileUploadPage/components/FileDropzone.tsx [hunk 1]
```diff
@@ -1,46 +1,157 @@
...
+    // check if the year is the first number in the date
+    fullDate = fName.match("[0-9]{4}([-/ .])[0-9]{2}[-/ .][0-9]{2}");
+    if (fullDate) {
+      const splitDate = fullDate[0].split(fullDate[1]);
+      return splitDate[0];
+    }
+    return undefined;
+  };
 
+  //accept the files that are dropped into the dropzone
```
Comment (ralphgregorio) [inline frontend/src/pages/FileUploadPage/components/FileDropzone.tsx]:
```suggestion
  // accept the files that are dropped into the dropzone
```

File: frontend/src/pages/FileUploadPage/components/FileDropzone.tsx [hunk 2]
```diff
@@ -1,46 +1,157 @@
...
+        if (fy[year]) fy[year].push(file);
+        else fy[year] = [file];
+        yearSet.add(year);
+      });
+
+      setFilesPerYear(fy);
+    }
+  };
+
+  //zip a years group of files then send them to the api
```
Comment (ralphgregorio) [inline frontend/src/pages/FileUploadPage/components/FileDropzone.tsx]:
```suggestion
  // zip a years group of files then send them to the api
```

File: frontend/src/pages/FileUploadPage/components/FileDropzone.tsx [hunk 3]
```diff
@@ -1,46 +1,157 @@
...
-      "application/xml": [".xml"],
-    },
+    accept: fileType,
   });
 
-  const acceptedFileItems = files.map((file: FileWithPath) => (
-    <li key={file.path}>
-      {file.path} - {file.size} bytes
-    </li>
+  //displays uploaded files grouped by year
```
Comment (ralphgregorio) [inline frontend/src/pages/FileUploadPage/components/FileDropzone.tsx]:
```suggestion
  // displays uploaded files grouped by year
```

**Comment by redwing14e** [comment]
complete

**Pull request merged**